src/main/java/io/reactivex/rxjava3/core/Scheduler.java

```diff
@@ -542,9 +540,10 @@ public void run() {
                 try {
                     run.run();
                 } catch (Throwable ex) {
-                    Exceptions.throwIfFatal(ex);
-                    worker.dispose();
-                    throw ExceptionHelper.wrapOrThrow(ex);
+                    // Exceptions.throwIfFatal(ex); nowhere to go
+                    dispose();
+                    RxJavaPlugins.onError(ex);
+                    throw ex;
```

Comment:
@JakeWharton Does this syntax work with Android desugar (rethrowing final `Throwable`s)?

Reply:
Hmm this looks like a normal throw. I don't understand what might be a compatibility concern here.

Reply:
The Java 7 compiler feature that let's you rethrow constant cheched or `Throwable` exceptions from within a `catch` block, even if the method didn't specify a `throws` clause.

Reply:
Ah, missed it was `Throwable`. Checked exceptions are only a feature of the Java compiler not the Java bytecode so yeah the Android toolchain won't care 👍 .